api/location/schema.py

```diff
@@ -154,6 +154,12 @@ class Query(graphene.ObjectType):
     def resolve_all_locations(self, info):
         query = Location.get_query(info)
         result = query.filter(LocationModel.state == "active")
+        for location in result:
```

Comment:
@mnswaleh instead of looping, what if we use something like `result.join(RoomModel, (Location.room_id == RoomModel.id) and (RoomModel.state == "active"))`?